cmd/revokeuser.go

```diff
@@ -112,77 +176,38 @@ func (p *RevokeUserParams) PreInteractive(ctx ActionCtx) error {
 	return nil
 }
 
-func (p *RevokeUserParams) Load(ctx ActionCtx) error {
-	var err error
-
-	if err = p.AccountContextParams.Validate(ctx); err != nil {
-		return err
-	}
-
-	if p.user == "" && p.userPubKey == "" {
-		n := ctx.StoreCtx().DefaultUser(p.AccountContextParams.Name)
-		if n != nil {
-			p.user = *n
-		}
-	}
-	if p.userPubKey == "" && p.user == "" {
-		return fmt.Errorf("user or user-public-key are required")
-	}
-	if p.userPubKey != "" && p.user != "" {
-		return fmt.Errorf("user and user-public-key are mutually exclusive")
-	}
-	p.claim, err = ctx.StoreCtx().Store.ReadAccountClaim(p.AccountContextParams.Name)
-	if err != nil {
-		return err
-	}
-	if p.userPubKey != "" {
-		return nil
-	}
-	userClaim, err := ctx.StoreCtx().Store.ReadUserClaim(p.AccountContextParams.Name, p.user)
-	if err != nil {
-		return err
-	}
-
-	if err != nil || userClaim == nil {
-		return fmt.Errorf("user is required")
-	}
-
-	p.userPubKey = userClaim.Subject
-
-	return nil
-}
-
 func (p *RevokeUserParams) Validate(ctx ActionCtx) error {
-	if !nkeys.IsValidPublicUserKey(p.userPubKey) {
-		return fmt.Errorf("user-public-key %q is not a valid public user nkey", p.userPubKey)
+	if p.userKey.publicKey == "" && p.user == "" {
+		return fmt.Errorf("user or user-public-key is required")
 	}
-	if err := p.SignerParams.Resolve(ctx); err != nil {
+	if err := p.userKey.Valid(); err != nil {
 		return err
 	}
-
-	return nil
-}
-
-func (p *RevokeUserParams) PostInteractive(ctx ActionCtx) error {
-	return nil
+	return p.SignerParams.Resolve(ctx)
 }
 
 func (p *RevokeUserParams) Run(ctx ActionCtx) (store.Status, error) {
 	if p.at == 0 {
-		p.claim.Revoke(p.userPubKey)
+		p.claim.Revoke(p.userKey.publicKey)
 	} else {
-		p.claim.RevokeAt(p.userPubKey, time.Unix(int64(p.at), 0))
+		p.claim.RevokeAt(p.userKey.publicKey, time.Unix(int64(p.at), 0))
 	}
-
 	token, err := p.claim.Encode(p.signerKP)
 	if err != nil {
 		return nil, err
 	}
-
 	r := store.NewDetailedReport(true)
 	StoreAccountAndUpdateStatus(ctx, token, r)
 	if r.HasNoErrors() {
-		r.AddOK("revoked user %s", p.userPubKey)
+		if p.userKey.publicKey == "*" {
+			when := int64(p.at)
+			if when == 0 {
+				when = time.Now().Unix()
+			}
+			r.AddOK("revoked all users issued before %s", time.Unix(when, 0).String())
```

Comment:
wouldn't we expect some cleanup as well?
if I already have revocations and add * makes all revocations for a user prior to when obsolete